File: internal/models/project.go
package models

type Project struct {
	ID           int                `json:"id"`
	Name         string             `json:"name"`
	Slug         string             `json:"slug"`
	Description  string             `json:"description"`
	LogoSmallURL string             `json:"logo_small_url"`
	LogoBigURL   string             `json:"logo_big_url"`
	Roles        []Role             `json:"roles"`
	Points       []Point            `json:"points"`
	USStatuses   []UserStoryStatus  `json:"us_statuses"`
	Swimlanes    []Swimlane         `json:"swimlanes"`
	Milestones   []ProjectMilestone `json:"milestones"`
	IAmOwner     bool               `json:"i_am_owner"`
	IAmAdmin     bool               `json:"i_am_admin"`
	IAmMember    bool               `json:"i_am_member"`
	// Members      []ProjectMember `json:"members"`

	MaxPoints    float64 `json:"max_points"`
	RolePoints   []int   `json:"role_points"`
	StatusPoints []int   `json:"status_points"`
}

type Swimlane struct {
	ID        int    `json:"id"`
	Name      string `json:"name"`
	Order     int    `json:"order"`
	ProjectID int    `json:"project"`
}

type ProjectMilestone struct {
	ID     int    `json:"id"`
	Name   string `json:"name"`
	Slug   string `json:"slug"`
	Closed bool   `json:"closed"`
}

type ProjectExtraInfo struct {
	ID           int    `json:"id"`
	Name         string `json:"name"`
	Slug         string `json:"slug"`
	LogoSmallURL string `json:"logo_small_url"`
}

File: frontend/src/models/project.ts
export interface Project {
    id: number;
    name: string;
    slug: string;
    description: string;
    logo_small_url: string;
    logo_big_url: string;
    roles: Role[];
    points: Point[];
    us_statuses: UserStoryStatus[];
    swimlanes: Swimlane[];
    // members: Member[];
}

export interface Role {
    id: number;
    name: string;
    slug: string;
    project_id: number;
    order: number;
    computable: boolean;
}

export interface Point {
    id: number;
    project_id: number;
    name: string;
    value: number;
}

export interface UserStoryStatus {
    id: number;
    project_id: number;
    name: string;
    color: string;
    slug: string;
    order: number;
    is_closed: boolean;
    is_archived: boolean;
}

export interface Swimlane {
    id: number;
    project_id: number;
    name: string;
    order: number;
}